### DISCUSSION TITLE
socket.io validation seemingly broken

### DISCUSSION THREAD

**Discussion by alexherrera42** [artilleryio/artillery#1778] [Artillery]
I copied and pasted the source for these files from your examples folder here on this github repo.  The reason I did this was because I can't seem to get even the most simple case of socket.io test to capture or validate any responses from socket.io.  After running these exact copies from your examples I have come to the conclusion that socket.io testing is broken in the current version of artillery.  I was running 2.0.0-27, and in this screen shot I have upgraded to 2.0.0-28.  I am running on windows under git bash with node 16.  socket.io is 4.5.4

![image](https://user-images.githubusercontent.com/107720780/213554836-186a525d-c740-4205-9695-c13b3a6ce72a.png)

**Comment by farhan888** [comment]
To validate a response, set a response object at the same level of emit.

**Comment by bernardobridge** [comment]
Just to add to the above, there was an issue with this functionality that was fixed with [version 2.0.0-38](https://github.com/artilleryio/artillery/releases/tag/v2.0.0-38).

Please update to a more recent version (we are currently on 2.0.5) when trying the suggestion above.
